### PR TITLE
Update README with code snippet info

### LINKED ISSUE #423: Update README with code snippets info
Code snippets extension has been included in 0.9 release. Even though it is still at experimental stage, we also need to update our docs with some info about the new feature.

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Fixes #423
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by karlaspuldaro** [comment]
I noticed our images and gifs are now outdated as they don't show code snippet extension. We can fix them all in this PR or create a new issue.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Elyra currently includes:
 * [Notebook versioning based on git integration](#notebook-versioning-based-on-git-integration)
 * [Notebook navigation using auto-generated **Table of Contents**](#notebook-navigation-using-auto-generated-table-of-contents)
 * [Reusable configuration for runtimes](#reusable-configuration-for-runtimes)
+* [Add custom Code Snippets - Experimental](#code-snippets)
```
Comment (lresende) [inline README.md:37]: I would move this to right after `Python script...` and would call it `Reusable code snippets (experimental)`

File: README.md [hunk 2]
```diff
@@ -105,6 +106,37 @@ Elyra introduces a 'shared configuration service' that simplifies workspace conf
 enabling things like external runtime access details to be configured once and shared
 across multiple components.  
 
+#### Add custom Code Snippets - Experimental
```
Comment (lresende) [inline README.md:109]: Title something like `Reusable code snippets (experimental)`

File: docs/dev/code-snippets.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		"code": [
+			"def is_even(num):",
+			"   return num % 2 == 0"
+		]
+	}
+}
+```
+NOTE: `code` field content must be a string array split by line.
+
+Once code snippet extension is in use, the application component retrieves data using **Elyra Metadata Service**
```
Comment (lresende) [inline docs/dev/code-snippets.md:48]: We have a doc for metadata, so link to it.

File: docs/dev/code-snippets.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# Elyra Code Snippets - Experimental
+
+The ability to reuse pieces of code allows users to avoid doing repetitive work, 
+making the programming workflow much more simple and productive.
+Elyra supports custom code snippets that can be added to the file editor.
+
+![Code Snippet Sample](../source/images/code-snippet-expanded.png)
+
+## Code Snippet data
```
Comment (lresende) [inline docs/dev/code-snippets.md:27]: Should we have a small paragraph here and do something more detailed down below similar to how we tell people to create runtime? (I am not sure what's best here... as this is becoming very large)

File: docs/dev/code-snippets.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	}
+}
+```
+NOTE: `code` field content must be a string array split by line.
+
+Once code snippet extension is in use, the application component retrieves data using **Elyra Metadata Service**
+through a REST API, and displays each snippet by `display_name` in the extension UI.
+Each snippet item can then be expanded on click, also displaying the `code` content.
+Currently, the application does not support full integration of adding the code into the editor by
+clicking the `+` button, although it does allow user to manually copy the snippet and paste it
```
Comment (kevin-bates) [inline docs/dev/code-snippets.md:52]: allow _the_ user

File: README.md [hunk 6]
```diff
@@ -84,6 +85,38 @@ them against local or cloud-based resources seamlessly.
...
+}
+```
+NOTE: code field content is split by line
+
+The extension then retrieves the data and displays a list of code snippets,
+which can be individually expanded on click for easy access of the code content.
+The snippet can be manually copied from the viewer into the editor.
+
+![Code Snippet Sample](docs/source/images/code-snippet-expanded.png)
+
```
Comment (kevin-bates) [inline README.md:119]: I still think this is too much information in the README.  I would suggest only showing the code-snippet-expanded image, followed by a statement like:  "For more information see Elyra Code Snippets" where "Elyra Code Snippets" is a link to code-snippets.html.

File: README.md [hunk 7]
```diff
@@ -92,31 +92,9 @@ Elyra supports a beta version of the **Code Snippet** feature.
...
-}
-```
-NOTE: code field content is split by line
-
-The extension then retrieves the data and displays a list of code snippets,
-which can be individually expanded on click for easy access of the code content.
-The snippet can be manually copied from the viewer into the editor.
-
-![Code Snippet Sample](docs/source/images/code-snippet-expanded.png)
+For more information on how to configure code snippets metadata see [Elyra Code Snippets](https://github.com/elyra-ai/elyra/blob/master/docs/dev/code-snippets.md)
```
Comment (kevin-bates) [inline README.md:97]: I think the target of this link should be `docs/dev/code-snippets.md` until we come up with a location for our docs.  If we choose `readthedocs`, this link will be something like...
```
[Elyra Code Snippets](https://elyra.readthedocs.io/en/latest/code-snippets.html)
```
despite the file being a `.md` file...
```
https://github.com/elyra-ai/elyra/blob/master/docs/dev/code-snippets.md
```

**Comment by ajbozarth** [comment]
Nit: you new `pipeline-editor-properties-menu.png` doesn't have `Properties` highlighted in the context menu like it did before

**Pull request merged**